astropy/coordinates/tests/test_name_resolve.py

```diff
@@ -155,7 +155,7 @@ def test_name_resolve_cache(tmpdir):
 
         urls = get_cached_urls()
         assert len(urls) == 1
-        assert 'http://cdsweb.u-strasbg.fr/cgi-bin/nph-sesame/' in urls[0]
+        assert any([urls[0].startswith(x) for x in sesame_url._value])
```

Comment:
```suggestion
        assert urls[0].rstrip('A?castor') in sesame_url._value
```
Both this and the generator version laid out by @nstarman are concise and readable in code, but I'd actually recommend to get back closer to your original idea. Reason being, the other options are giving rather unhelpful error messages in case of a failure, like
```python
>           assert any([urls[0].startswith(x[1:]) for x in sesame_url._value])
E           assert False
E            +  where False = any([False, False])
```

Reply:
Well, can we guarantee the URL query always uses `A?` ? I guess I can try this suggestion and see what happens. You should also try in Europe to see if that actually fixes your problem. I will push a commit out shortly. Thanks!

Reply:
The standard scheme seems to be
`http://cdsweb.u-strasbg.fr/cgi-bin/nph-sesame/A?<target>`
(from `de` domain, but whether `vizier.cfa.harvard.edu` is used is apparently non-deterministic).
But you might also use something like `urls[0][:44]`, which should be sufficiently specific.

Reply:
`A?` seems to pass the remote data CI. Should I leave it at that, or do you wish to not hardcode `A?`?

Reply:
Hmm, `A` is simply the database initial (here "all", but I guess it might in theory then take a different value) – perhaps the most robust implementation is `urls[0][:-8]`.

Reply:
Hardcoding any slice is just as dangerous. I'll think of a way using `startswith` that won't cause confusing error.

Reply:
Possibly, though the target name is given directly in the test, and I suppose the construction as `"{db}?{name}"` is unlikely to change (could use `urls[0][:-len(target_name)-2]`).

Reply:
Good solution in 57ac641!